src/runtime/ArgumentsObject.cpp

```diff
@@ -347,30 +347,31 @@ void ArgumentsObject::setIndexedPropertyValueQuickly(ExecutionState& state, uint
 
 bool ArgumentsObject::isModifiedArgument(uint64_t index)
 {
-    if (m_modifiedArguments.size() == 0) {
+    if (!modifiedArguments()) {
         return false;
     }
-    if (LIKELY(index != Value::InvalidIndexValue) && index < m_argc) {
-        return m_modifiedArguments[index];
+    if (LIKELY(index != Value::InvalidIndexValue) && index < argc()) {
+        return m_modifiedArguments->m_modified[index];
     }
     return false;
 }
 
 void ArgumentsObject::setModifiedArgument(uint64_t index)
 {
     if (LIKELY(index != Value::InvalidIndexValue)) {
-        if (m_modifiedArguments.size() == 0) {
-            m_modifiedArguments.resize(m_argc, false);
+        if (modifiedArguments() == nullptr) {
+            size_t c = argc();
+            m_modifiedArguments = new ModifiedArguments(c);
         }
         if (index < m_argc) {
```

Comment:
I think that `m_argc` should be replaced by `argc()` at this point because `m_argc` is rewritten by `ModifiedArguments`.

Reply:
oops!

Reply:
I fixed the problem